### ISSUE TITLE
Attrs - change file attributes doesn't work

### ISSUE THREAD

**Issue by 1ricardo** [saltstack/salt#48401]
### Description of Issue/Question
Hello,
I noticed that the functionality of **file.managed -> attrs** doesn't work at all.

According with documentations on https://docs.saltstack.com/en/latest/ref/states/all/salt.states.file.html#salt.states.file.managed

`attrs -  The attributes to have on this file, e.g. a, i. The attributes can be any or a combination of the following characters: acdijstuADST.` 

### Setup
Nothing special was added to setup, just standard installation 

### Steps to Reproduce Issue
I prepared simply test, I want to set the immutable flag for `/etc/resolv.conf`:

##### This is list of attribute before executing the Salt SLS file
```sh
~ # lsattr /etc/resolv.conf
---------------- /etc/resolv.conf

```
##### This is content of test.sls file to change the attribute
```sh
resolve_change_ATTR:
  file.managed:
    - name: /etc/resolv.conf
    - user: root
    - group: root
    - mode: 644
    - attrs: i
```
##### This is debug from executing the SLS file
```sh
~ # salt-call state.sls consul.test  -l debug
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: web159a-stg-1.omarsys.inet
[DEBUG   ] Configuration file path: /etc/salt/minion
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Loading static grains from /etc/salt/grains
[DEBUG   ] Connecting to master. Attempt 1 of 1
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506')
[DEBUG   ] Generated random reconnect delay between '1000ms' and '11000ms' (9115)
[DEBUG   ] Setting zmq_reconnect_ivl to '9115ms'
[DEBUG   ] Setting zmq_reconnect_ivl_max to '11000ms'
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506', 'clear')
[DEBUG   ] Decrypting the current master AES key
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] Determining pillar cache
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506')
[DEBUG   ] Loaded minion key: /etc/salt/pki/minion/minion.pem
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] LazyLoaded state.sls
[DEBUG   ] LazyLoaded saltutil.is_running
[DEBUG   ] LazyLoaded grains.get
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506')
[INFO    ] Loading fresh modules for state activity
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] In saltenv 'base', looking at rel_path 'consul/test.sls' to resolve 'salt://consul/test.sls'
[DEBUG   ] In saltenv 'base', ** considering ** path '/var/cache/salt/minion/files/base/consul/test.sls' to resolve 'salt://consul/test.sls'
[DEBUG   ] Fetching file from saltenv 'base', ** attempting ** 'salt://consul/test.sls'
[DEBUG   ] No dest file found
[INFO    ] Fetching file from saltenv 'base', ** done ** 'consul/test.sls'
[DEBUG   ] compile template: /var/cache/salt/minion/files/base/consul/test.sls
[DEBUG   ] Jinja search path: ['/var/cache/salt/minion/files/base']
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/consul/test.sls' using 'jinja' renderer: 0.0027449131012
[DEBUG   ] Rendered data from file: /var/cache/salt/minion/files/base/consul/test.sls:
resolve_change_ATTR:
  file.managed:
    - name: /etc/resolv.conf
    - user: root
    - group: root
    - mode: 644
    - attrs: i

[DEBUG   ] LazyLoaded config.get
[DEBUG   ] Results of YAML rendering: 
OrderedDict([('resolve_change_ATTR', OrderedDict([('file.managed', [OrderedDict([('name', '/etc/resolv.conf')]), OrderedDict([('user', 'root')]), OrderedDict([('group', 'root')]), OrderedDict([('mode', 644)]), OrderedDict([('attrs', 'i')])])]))])
[PROFILE ] Time (in seconds) to render '/var/cache/salt/minion/files/base/consul/test.sls' using 'yaml' renderer: 0.00557398796082
[DEBUG   ] LazyLoaded file.managed
[INFO    ] Running state [/etc/resolv.conf] at time 12:05:06.640215
[INFO    ] Executing state file.managed for [/etc/resolv.conf]
[WARNING ] State for file: /etc/resolv.conf - Neither 'source' nor 'contents' nor 'contents_pillar' nor 'contents_grains' was defined, yet 'replace' was set to 'True'. As there is no source to replace the file with, 'replace' has been set to 'False' to avoid reading the file unnecessarily.
[DEBUG   ] LazyLoaded file.user_to_uid
[INFO    ] File /etc/resolv.conf exists with proper permissions. No changes made.
[INFO    ] Completed state [/etc/resolv.conf] at time 12:05:06.661756 duration_in_ms=21.54
[DEBUG   ] File /var/cache/salt/minion/accumulator/140708586415568 does not exist, no need to cleanup.
[DEBUG   ] LazyLoaded config.option
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506', 'aes')
[DEBUG   ] Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'web159a-stg-1.inet', 'tcp://10.200.12.78:4506')
[DEBUG   ] LazyLoaded highstate.output
local:
----------
          ID: resolve_change_ATTR
    Function: file.managed
        Name: /etc/resolv.conf
      Result: True
     Comment: File /etc/resolv.conf exists with proper permissions. No changes made.
     Started: 12:05:06.640216
    Duration: 21.54 ms
     Changes:   

Summary for local
------------
Succeeded: 1
Failed:    0
------------
Total states run:     1
Total run time:  21.540 ms
```

As you can see the flag was not set
##### After executing the state
```sh
~ # lsattr /etc/resolv.conf
---------------- /etc/resolv.conf
```

I can change the attribute manually `chattr +i /etc/resolv.conf`, and this is how it should looks like:
```sh
~ # lsattr /etc/resolv.conf
----i----------- /etc/resolv.conf
```

### Versions Report
```sh
~ # salt-call --versions-report
Salt Version:
           Salt: 2017.7.4
 
Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: 0.21.1
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.1
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Apr 11 2018, 07:36:10)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4
 
System Versions:
           dist: centos 7.5.1804 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-693.21.1.el7.x86_64
         system: Linux
        version: CentOS Linux 7.5.1804 Core
```

**Comment by Ch3LL** [comment]
you are looking at the latest docs and not 2017.7 which is the version you are using. This argument was added in this PR: https://github.com/saltstack/salt/pull/41554
which was included in 2018.3.0 and above. If you try the latest 2018.3 release does it work?

**Comment by 1ricardo** [comment]
Hi @Ch3LL 

Thank you for your involve in that case. I am sorry for my mistake, I missed the information, that it was implemented in version 2018.3  

I made the test on **salt 2018.3.2 (Oxygen)** and all works fine
